src/main/java/de/tum/in/www1/artemis/repository/FileUploadSubmissionRepository.java

```diff
@@ -1,15 +1,59 @@
 package de.tum.in.www1.artemis.repository;
 
+import java.util.Optional;
+
+import org.springframework.data.jpa.repository.EntityGraph;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
 import de.tum.in.www1.artemis.domain.FileUploadSubmission;
 
 /**
  * Spring Data JPA repository for the FileUploadSubmission entity.
  */
-@SuppressWarnings("unused")
 @Repository
 public interface FileUploadSubmissionRepository extends JpaRepository<FileUploadSubmission, Long> {
 
+    /**
+     * @param courseId the course id we are interested in
+     * @return the number of submissions belonging to the course id, which have the submitted flag set to true and the submission date before the exercise due date, or no exercise
+     *         due date at all
+     */
+    @Query("SELECT COUNT (DISTINCT submission) FROM FileUploadSubmission submission WHERE submission.participation.exercise.course.id = :#{#courseId} AND submission.submitted = TRUE AND (submission.submissionDate < submission.participation.exercise.dueDate OR submission.participation.exercise.dueDate IS NULL)")
```

Comment:
The repository methods are copy and pasted from other repositories, e.g. ModelingSubmissionRepository.
Especially from a consistency approach this is not a good way to handle this as somebody might change one of the other repositories but not this one.

I would propose to refactor that you just add these methods into the SubmissionRepository and use the service to  to identify if the found submission has the correct class type (normally an edge case). 
This should then be streamlined in another PR with the other repositories.

Reply:
I did it for methods that return count, but I am not sure how to properly do it for ones that return `FileUploadSubmission` type.  I found out an article about JPA Query [SPEL expressions ](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#jpa.query.spel-expressions). We probably can use this idea, but then we need to change how `SubmissionRepository` is used for all services